lib/takeaway.rb

```diff
@@ -0,0 +1,51 @@
+require_relative 'menu'
+require_relative 'order'
+require_relative 'messenger'
+
+class Takeaway
+  attr_reader :menu, :messenger
+
+  def initialize(menu = Menu.new, messenger)
+    @menu = menu
+    @messenger = messenger
+    @current_order = Order.new
+  end
+
+  def view_menu
+    @menu.dishes
+  end
+
+  def add_to_order(item,quantity)
+    order = current_order
+    dish = @menu.dishes.select { |k,v| k == item }
```

Comment:
This feels like it's breaking the encapsulation of `@menu` a bit because it's reaching in to get `dishes` and then making assumptions about what `dishes` is and how it should be used.  Try moving any code that works on `dishes` inside `Menu`.  eg you could refactor this line to call something like `@menu.dish(item)`.